src/components/LooMap/useLocateMapControl.ts

```diff
@@ -21,35 +32,33 @@ const LocationMarker = L.Marker.extend({
       html: svg,
       iconSize: [diameter, diameter],
     });
-  },
+  }
 
-  createIcon: function () {
+  private createIcon() {
     let style = '';
 
-    const styleOptions = {
+    const styleOptions: { [key: string]: string } = {
       color: 'stroke',
       weight: 'stroke-width',
       fillColor: 'fill',
       fillOpacity: 'fill-opacity',
       opacity: 'opacity',
     };
 
-    // convert style options to css string
     Object.entries(styleOptions).forEach(([option, property]) => {
-      if (this.options[option]) {
-        style = style + `${property}: ${this.options[option]};`;
+      const value = this.options[option as keyof LocationMarkerOptions];
+      if (value !== undefined) {
```

Comment:
What if the value is numeric? Should we add a `px` suffix to make it a valid CSS length?